code/controllers/subsystem/polling.dm

```diff
@@ -0,0 +1,312 @@
+
+/datum/controller/subsystem/polling/proc/poll_ghost_candidates_for_mob(question, role, check_jobban, poll_time = 30 SECONDS, mob/target_mob, ignore_category = null, flashwindow = TRUE, pic_source, role_name_text)
+	var/static/list/mob/currently_polling_mobs = list()
+
+	if(currently_polling_mobs.Find(target_mob))
+		return list()
+
+	currently_polling_mobs += target_mob
+
+	var/list/possible_candidates = poll_ghost_candidates(question, role, check_jobban, poll_time, ignore_category, flashwindow, pic_source, role_name_text)
+
+	currently_polling_mobs -= target_mob
+	if(!target_mob || QDELETED(target_mob) || !target_mob.loc)
+		return list()
+
+	return possible_candidates
+
+/datum/controller/subsystem/polling/proc/poll_ghost_candidates_for_mobs(question, role, check_jobban, poll_time = 30 SECONDS, list/mobs, ignore_category = null, pic_source, role_name_text)
+	var/list/candidate_list = poll_ghost_candidates(question, role, check_jobban, poll_time, ignore_category, pic_source, role_name_text)
+
+	for(var/mob/potential_mob as anything in mobs)
+		if(QDELETED(potential_mob) || !potential_mob.loc)
+			mobs -= potential_mob
+
+	if(!length(mobs))
+		return list()
+
+	return candidate_list
+
+/datum/controller/subsystem/polling/proc/is_eligible(mob/M, role, check_jobban, the_ignore_category)
+	if(!M.key || !M.client)
+		return FALSE
+	if(the_ignore_category)
+		if(M.ckey in GLOB.poll_ignore[the_ignore_category])
+			return FALSE
+	if(role)
+		if(!(role in M.client.prefs.be_special))
+			return FALSE
+		var/required_time = GLOB.special_roles[role] || 0
+		if(M.client && M.client.get_remaining_days(required_time) > 0)
+			return FALSE
+
+	if(check_jobban)
+		if(is_banned_from(M.ckey, list(check_jobban, ROLE_SYNDICATE)))
+			return FALSE
+
+	return TRUE
+
+/datum/controller/subsystem/polling/proc/polling_finished(datum/candidate_poll/P)
+	// Trim players who aren't eligible anymore
+	var/len_pre_trim = length(P.signed_up)
+	P.trim_candidates()
+	log_game("Candidate poll [P.role ? "for [P.role]" : "\"[P.question]\""] finished. [len_pre_trim] players signed up, [length(P.signed_up)] after trimming")
+
+	P.finished = TRUE
+	currently_polling -= P
+
+	// Determine which is the next poll closest the completion or "disable" firing if there's none
+	if(!length(currently_polling))
+		polls_active = FALSE
+		next_poll_to_finish = null
+	else if(P == next_poll_to_finish)
+		next_poll_to_finish = null
+		for(var/poll in currently_polling)
+			var/datum/candidate_poll/P2 = poll
+			if(!next_poll_to_finish || P2.time_left() < next_poll_to_finish.time_left())
+				next_poll_to_finish = P2
+
+/datum/controller/subsystem/polling/stat_entry(msg)
+	msg += "Active: [length(currently_polling)] | Total: [total_polls]"
+	if(next_poll_to_finish)
+		msg += " | Next: [DisplayTimeText(next_poll_to_finish.time_left())] ([length(next_poll_to_finish.signed_up)] candidates)"
+	return ..()
+
+// The datum that describes one instance of candidate polling
+/datum/candidate_poll
```

Comment:
i think this should be moved to it's own file in the `/datum` folder instead of being in the same file as the SS.